rustls-libssl/src/error.rs

```diff
@@ -142,6 +155,13 @@ impl Error {
         }
         self
     }
+
+    /// `WouldBlock` errors never make it on the error stack.
+    ///
+    /// They are usual in the use of non-blocking BIOs.
+    fn quiet(&self) -> bool {
+        matches!(self.reason, Reason::WouldBlock)
+    }
```

Comment:
WDYT about making `Error` derive `Eq` and `PartialEq`? I don't think there's anything preventing that today, it seems convenient, and it would allow replacing this with

```rust
    fn quiet(&self) -> bool {
        self.reason == Reason::WouldBlock
    }
 ```